src/plots/cartesian/autorange.js

```diff
@@ -173,15 +189,66 @@ function makePadFn(ax) {
     return function getPad(pt) { return pt.pad + (pt.extrapad ? extrappad : 0); };
 }
 
-function doAutoRange(ax) {
+function concatExtremes(gd, ax, ext) {
+    var i;
+    var out = [];
+
+    var fullData = gd._fullData;
+
+    for(i = 0; i < fullData.length; i++) {
+        var trace = fullData[i];
+        var extremes = trace._extremes;
+
+        if(trace.visible === true) {
+            if(Registry.traceIs(trace, 'cartesian') || Registry.traceIs(trace, 'gl2d')) {
+                var axId = ax._id;
+                if(extremes[axId]) {
+                    out = out.concat(extremes[axId][ext]);
```

Comment:
In principle we can collapse `out` more than just `concat` - new items can be mooted or can moot existing items based on the [three criteria used in `expand`/`findExtremes`](https://github.com/plotly/plotly.js/pull/2849/files#diff-a701daa267f7b5c5bce68efe85034854R418)

I suppose it's worth testing with some real examples though - `doAutoRange` scales as `O(min.length * max.length)` which says it's a good idea to minimize the length of the min and max arrays, but excluding items from the `min` and `max` arrays scales as something like `O(finalArray.length * uncompressedTotalLength)` which is also sort of quadratic... not exactly a clear win either way.

Reply:
Actually there’s a probably much more common case where I’m pretty sure it’s a big win to collapse more here: lots of traces each with only one or two min/max entries. Like 100 line or bar traces, or 100 marker traces with constant marker size. Collapsing here takes ~100 * 2 comparisons, then `doAutorange` only has one calculation to do, whereas simple concat here leaves 100 * 100 calculations for `doAutorange`